sdk/core/Azure.Core.TestFramework/src/Instrumentation/InstrumentResultInterceptor.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Castle.DynamicProxy;
+
+namespace Azure.Core.TestFramework
+{
+    internal class InstrumentResultInterceptor : IInterceptor
+    {
+        private static readonly MethodInfo InstrumentOperationInterceptorMethodInfo = typeof(InstrumentResultInterceptor).
+            GetMethod(nameof(InstrumentOperationInterceptor), BindingFlags.NonPublic | BindingFlags.Instance);
```

Comment:
Same here about `.` and throwing with a more obvious reason in case this ever changes.